File: bankingOnGolang/internal/model/KycProfile.go
package model

import (
	"time"
)

// ==============================================================
// KycProfile Declaration
// ==============================================================
type KycProfile struct {
	BaseModel
	ProfileId         string
	LastReviewedOn    time.Time
	CustomerId        *uint
	Customer          *Customer          `gorm:"foreignKey:CustomerId"`
	IdentityDocuments []IdentityDocument `gorm:"foreignKey:IdentityDocumentsFromKycProfileId"`
	RiskAssessments   []RiskAssessment   `gorm:"foreignKey:RiskAssessmentsFromKycProfileId"`
	Screenings        []ScreeningResult  `gorm:"foreignKey:ScreeningsFromKycProfileId"`
	Status            KycStatus

	// parent associations as their child
	KycProfilesFromCustomerId *uint
}

File: src/app/models/KycProfile.ts


// Define collection and schema for KycProfile
export interface KycProfile {
    profileId: string
    lastReviewedOn: Date
    Customer: Schema.Types.ObjectId
    IdentityDocuments:  Schema.Types.ObjectId[]
    RiskAssessments:  Schema.Types.ObjectId[]
    Screenings:  Schema.Types.ObjectId[]
    Status:  String
    collection: 'kycProfiles'
}
